content/departments/cloud/technical-docs/v2.0/disaster_recovery_process.md

```diff
@@ -0,0 +1,119 @@
+export SLUG=<SLUG>
+export GKE_NAME=$(mi2 instance get -e $ENVIRONMENT --slug $SLUG | jq -r '.status.gkeClusters[0].name')
+export GKE_REGION=$(mi2 instance get -e $ENVIRONMENT --slug $SLUG | jq -r '.spec.gcpRegion')
+```
+
+- check instance is healthy
+
+```sh
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health
+curl -sSL --fail https://$SLUG.sourcegraph.com/sign-in -i
+```
+
+- connect to cluster
+
+```sh
+mi2 instance workon -e $ENVIRONMENT --slug $SLUG
+# paste output into terminal
+```
+
+- verify node zone
+
+```sh
+kubectl get nodes
+kubectl describe node <NODE_FROM_CLUSTER> | grep zone
+```
+
+- perform zone failover (remove node zone from GKE node locations)
+
+```sh
+gcloud container node-pools list --cluster $GKE_NAME --region $GKE_REGION
+gcloud container node-pools describe primary --cluster $GKE_NAME --region $GKE_REGION --format json | jq '.locations'
+gcloud container node-pools update primary --cluster $GKE_NAME --region $GKE_REGION --node-locations <DIFFERENT_ZONE_THAN_EXISTING_NODE> --async
+```
+
+- verify pods were terminated
+
+```
+kubectl get pods # should show failing pods, b/c node was drained
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health # should fail
+```
+
+- wait for new node to be ready
+
+```sh
+kubect get nodes # waiting for new node
+```
+
+- verify new node zone
+
+```sh
+kubectl describe node <NEW_NODE> | grep zone # should be different from previous node
+```
+
+- check instance is healthy
+
+```sh
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health
+curl -sSL --fail https://$SLUG.sourcegraph.com/sign-in -i
+```
+
+2. CloudSQL zone failover
+
+- export environment variables
+
+```sh
+export ENVIRONMENT=[dev|prod]
+export SLUG=<SLUG>
+export CLOUDSQL_INSTANCE_NAME=$(mi2 instance get -e $ENVIRONMENT --slug $SLUG | jq -r '.status.cloudSQL[0].name')
+export GCP_PROJECT=$(mi2 instance get -e $ENVIRONMENT --slug $SLUG | jq -r '.status.gcpProjectId')
+export INSTANCE_ID=$(mi2 instance get -e $ENVIRONMENT --slug $SLUG | jq -r '.metadata.name')
+```
+
+- check instance is healthy
+
+```sh
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health
+curl -sSL --fail https://$SLUG.sourcegraph.com/sign-in -i
+```
+
+- patch CloudSQL instance to use different zone
+
+```sh
+gcloud sql instances describe $CLOUDSQL_INSTANCE_NAME --project $GCP_PROJECT | grep zone
+gcloud sql instances patch $CLOUDSQL_INSTANCE_NAME --zone <NEW_ZONE> --project $GCP_PROJECT --async
+# this make take upt to 5 mins, instance is not searchable
+gcloud sql instances describe $CLOUDSQL_INSTANCE_NAME --project $GCP_PROJECT | grep zone
+# should return <NEW_ZONE>
+```
+
+- check instance is healthy
+
+```sh
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health
+curl -sSL --fail https://$SLUG.sourcegraph.com/sign-in -i
+```
+
+- restore instance via terraform
+
+```sh
+cd environments/$ENVIRONMENT/deployments/$INSTANCE_ID/terraform/stacks/sql
+terraform init && terraform apply
+```
```

Comment:
wouldn't this step override the previous command `gcloud sql instances patch $CLOUDSQL_INSTANCE_NAME --zone <NEW_ZONE> --project $GCP_PROJECT --async` and the instance is moved back to the previous zone again?

I think we're missing a step to update the new SQL location in terraform.

also, we may need a way to override the SQL zone independent of the global zone config in config.yaml

https://github.com/sourcegraph/controller/blob/a4da3484d07e661429b618504437c89b94d19fc4/internal/resource/sql/sql.go#L139-L141

Reply:
@michaellzc I am not sure if I understand - you mean to 'equal' terraform state after failover was performed? or do failover via terraform modification instead of gcloud command?

Reply:
the current failover process:
-  ✅ use `gcloud patch` to do zone failover
- ❌  run `terraform apply` to catch up on the state, this would have overwritten the previous command

my proposal

follow the break glass process https://handbook.sourcegraph.com/departments/cloud/technical-docs/v2.0/break_glass_process/

- modify `config.yaml` to set a new zone for sql
- mi2 generate cdktf
- `terraform apply` locally

Reply:
@michaellzc I followed the process you suggested via cdktf - updated the handbook. Also added restore from backup as last step, b/c when zone is completly down, this might be required and we have it tested in different zone as well.